pyarr/readarr.py

```diff
@@ -0,0 +1,580 @@
+        book ID if one is found.
+
+        Args:
+            id_ (int, optional): Database id for book. Defaults to None.
+
+        Returns:
+            JSON: Array
+        """
+        path = f"book/{id_}" if id_ else "book"
+        return self.request_get(path, self.ver_uri)
+
+    # POST /book
+
+    def add_book(
+        self,
+        db_id,
+        book_id_type,
+        root_dir,
+        quality_profile_id=None,
+        metadata_profile_id=None,
+        monitored=True,
+        search_for_new_book=False,
+        author_monitor="all",
+        author_search_for_missing_books=False,
+    ):
+        """Adds a new book and  its associated author (if not already added)
+
+        Args:
+            db_id (int): goodreads, isbn, asin ID for the book
+            book_id_type (str): goodreads / isbn / asin
+            root_dir (str): Directory for book to be stored
+            quality_profile_id (int, optional): quality profile id. Defaults to 1.
+            metadata_profile_id (int, optional): metadata profile id. Defaults to 0.
+            monitored (bool, optional): should the book be monitored. Defaults to True.
+            search_for_new_book (bool, optional): search for the book to download now. Defaults to False.
+            author_monitor (str, optional): monitor the author for new books. Defaults to "all".
+            author_search_for_missing_books (bool, optional): search for missing books from this author. Defaults to False.
+
+        Raises:
+            ValueError: error raised if book_id_type is incorrect
+
+        Returns:
+            JSON: Array
+        """
+        book_id_types = ["goodreads", "isbn", "asin"]
+        if book_id_type not in book_id_types:
+            raise ValueError(f"Invalid book id type. Expected one of: {book_id_types}")
+
+        book_json = self._book_json(
+            db_id,
+            book_id_type,
+            root_dir,
+            quality_profile_id,
+            metadata_profile_id,
+            monitored,
+            search_for_new_book,
+            author_monitor,
+            author_search_for_missing_books,
+        )
+        path = "book"
+        return self.request_post(path, self.ver_uri, data=book_json)
+
+    # DELETE /book/{id}
+    def del_book(self, id_, delete_files=False, import_list_exclusion=True):
+        """Delete the book with the given id
+
+        Args:
+            id_ (int): Database id for book
+            delete_files (bool, optional): If true book folder and files will be deleted. Defaults to False.
+            import_list_exclusion (bool, optional): Add an exclusion so book doesn't get re-added. Defaults to True.
+
+        Returns:
+            JSON: {}
+        """
+        params = {
+            "deleteFiles": delete_files,
+            "addImportListExclusion": import_list_exclusion,
+        }
+        path = f"book/{id_}"
+        return self.request_del(path, self.ver_uri, params=params)
+
```

Comment:
A couple of functions below here are missing the comment above stating the HTTP request (ex. `GET /book`).

I personally starting doing that to help with a quick glance on the HTTP request, to ensure all endpoints are developed and sort endpoints into sections of the code. Although its helpful, it's technically not needed.

You'll need to decide whether to keep this format or remove it. If you decide to keep it, you'll also need to reorganize the functions in this file.

Reply:
I started doing that then stopped, but I do think you are right it makes it easier to ensure everything is captured, I will also tidy this up.